epictrack-web/src/components/workPlan/status/StatusContext.tsx

```diff
@@ -32,37 +48,53 @@ export const StatusProvider = ({
   const [showStatusForm, setShowStatusForm] = React.useState<boolean>(false);
   const [showApproveStatusDialog, setShowApproveStatusDialog] =
     React.useState<boolean>(false);
+  const [isCloning, setIsCloning] = React.useState<boolean>(false);
   const [status, setStatus] = React.useState<Status>();
+  const query = useSearchParams<StatusContainerRouteParams>();
+  const workId = React.useMemo(() => query.get("work_id"), [query]);
+  const { getWorkStatuses } = useContext(WorkplanContext);
+  const { groups } = useAppSelector((state) => state.user.userDetail);
 
   const onDialogClose = () => {
     setShowStatusForm(false);
+    setIsCloning(false);
+  };
+
+  const isEditable = (is_approved: boolean) => {
+    return !is_approved || groups.includes("Super User");
   };
 
   const onSave = async (data: any, callback: () => any) => {
+    const { description, posted_date, is_approved } = data;
     try {
-      if (status) {
-        // TODO update or create a status
-        // const result = await statusService?.update(data, status.id.toString());
-        // if (result && result.status === 200) {
-        //   showNotification(`${status.title} details updated`, {
-        //     type: "success",
-        //   });
-        //   setStatus(undefined);
-        //   setBackdrop(false);
-        //   callback();
-        // }
+      if (status && !isCloning && isEditable(is_approved)) {
+        const result = await statusService?.update(
+          Number(workId),
+          Number(status.id),
+          { description, posted_date }
+        );
+        if (result && result.status === 200) {
```

Comment:
I'm not sure whether we need to explicitly check whether the result is successful. Axios throws an error if it weren't and that will be picked up by the catch